src/Core/src/Platform/Android/WebViewExtensions.cs

```diff
@@ -38,6 +38,17 @@ public static void UpdateSettings(this AWebView platformWebView, IWebView webVie
 			platformWebView.Settings.DomStorageEnabled = domStorageEnabled;
 		}
 
+		public static void UpdateUserAgent(this AWebView platformWebView, IWebView webView)
+		{
+			if (platformWebView.Settings == null)
+				return;
+
+			if (webView.UserAgent != null)
+				platformWebView.Settings.UserAgentString = webView.UserAgent;
+			else
+				webView.UserAgent = platformWebView.Settings.UserAgentString;
```

Comment:
Thinking aloud here: What happens if the user sets the webview.UserAgent to null? Will it just read back from the control? Is it ever possible to have a "null user agent"? And a empty/"" UA?

Reply:
I do see this in the docs:
> If the parameter is "null" the User Agent will not be updated and the current User Agent will remain.

Is that intentional or a side effect?

Reply:
> Thinking aloud here: What happens if the user sets the webview.UserAgent to null? Will it just read back from the control? Is it ever possible to have a "null user agent"? And a empty/"" UA?

Most platforms document the behavior of a `null` or empty User Agent string and maps it with a platform specific behavior.

For [WinUI's WebView2](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2settings.useragent?view=webview2-dotnet-1.0.1343.22):
> If the parameter is empty the User Agent will not be updated and the current User Agent will remain.

For [Android's WebView](https://developer.android.com/reference/android/webkit/WebSettings#setUserAgentString(java.lang.String)):
> If the string is null or empty, the system default value will be used.

For [iOS's WKWebView](https://developer.apple.com/documentation/webkit/wkwebview/1414950-customuseragent)
> Use this property to specify a custom user agent string for your web view. The default value of this property is nil.
(So if it is `null` the "default" User Agent string will be used).

So technically it's not possible to have a `null` User Agent. The underlying controls would just use the existing or system default value.

The piece of code you reviewed, as well as its counterparts on other platforms, serves both as a function for updating the platform native control and initializing the MAUI control. When the MAUI control's user agent is null, this should mean that the user agent is uninitialized. It therefore attempts to initialize the value from the native control. This behavior is useful for platforms where the WebView does not expose its properties until it is fully loaded, such as WinUI, where the `CoreWebView2Settings.UserAgent` is only available after [`WebView2.CoreWebView2Initialized`](https://learn.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.webview2.corewebview2initialized?view=winui-2.8) event.

> If the parameter is "null" the User Agent will not be updated and the current User Agent will remain.
This is intentional. Because setting the User Agent to null will cause different behaviors in each platform, for a consistent cross-platform experience something should be done to standardize this in MAUI.

I therefore chose the WinUI behavior (as you can see the string is copied from WebView2's documentation). The behavior can easily be enforced in MAUI's WebView code: The setter for `UserAgent` actively blocks `null`s by setting the value to `value ?? UserAgent`.